common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerFactory.scala

```diff
@@ -47,6 +47,8 @@ case class ContainerPoolConfig(userMemory: ByteSize, concurrentPeekFactor: Doubl
   def cpuShare(reservedMemory: ByteSize) = (totalShare / (userMemory.toBytes / reservedMemory.toBytes)).toInt
 }
 
+case class RuntimesRegistryConfig(url: String, user: Option[String] = None, password: Option[String] = None)
```

Comment:
Should the credentials be their own type which should then be optional? It can get awkward if you have possibly two optional types but do you really want the case of `Config("bla", None, Some("password"))`? I'd assume no?

Reply:
Makes totally sense 👍